HB/instance.elpi

```diff
@@ -73,8 +73,10 @@ declare-instances T [class Class Struct MLwP|Rest] [CS|L] :-
 
   !,
   coq.term->gref T TGR,
+  coq.gref->path TGR TPath,
+  std.rev TPath [TMod|_],
   coq.gref->id TGR TID,
-  Name is TID ^ "_is_a_" ^ {gref->modname Struct},
+  Name is TMod ^ "_" ^ TID ^ "_canonical_" ^ {gref->modname Struct},
```

Comment:
Does this always work? For concrete instances it seems odd.
What does it do for, say, `prod`?

Reply:
Yes, it always work, the name of the current file will serve as the module name when defined at toplevel, and for `prod`, I guess it's `Datatypes`. (the CI passes and it contains many instances on `prod`, I did not check the outcome though)